simulator/lang/lib.mjs

```diff
@@ -0,0 +1,264 @@
+    return await c.getStates();
+  }
+
+  async getCurrentActor() {
+    const l = await c.getStateLocals(this.stateID)
+    if (l.l_curr_actor_id === consensusID) {
+      return this.consensus
+    } else {
+      return this.participants[l.l_curr_actor_id]
+    }
+  }
+
+  async newAccount() {
+    return await c.newAccount(this.stateID);
+  }
+
+  async newToken() {
+    return await c.newToken(this.stateID);
+  }
+}
+
+class Actor {
+
+  async getNextAction() {
+    const act = await c.getActions(this.scene.stateID,this.id)
+    return new Action(act[0],act[1].tag,this,this.scene);
+  }
+
+  async getStore() {
+    const l = await c.getStateLocals(this.scene.stateID)
+    return l.l_locals[this.id].l_store
+  }
+
+  async getWallet() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_ledger[this.account.id]
+  }
+
+
+  async getPhase() {
+    const l = await c.getStateLocals(this.scene.stateID)
+    return l.l_locals[this.id].l_phase
+  }
+
+  async getStatus() {
+    const l = await c.getStateLocals(this.scene.stateID)
+    switch (l.l_locals[this.id].l_ks) {
+      case 'PS_Suspend':
+        return 'Running';
+      case 'PS_Done':
+        return 'Done';
+    }
+  }
+
+  async history() {
+    const sg = await c.getStateGraph();
+    const filtered = Object.filter(sg, ([id, state]) =>
+      state[1].contents.l_curr_actor_id === this.id
+    );
+    return filtered;
+  }
+
+}
+
+class Participant extends Actor {
+  constructor(id,account,name,scene) {
+    super();
+    this.id = id;
+    this.account = account;
+    this.name = name;
+    this.scene = scene;
+  }
+
+  async init(liv={},accID="") {
+    const r = await c.initFor(this.scene.stateID,this.id,JSON.stringify(liv),accID)
+    ++this.scene.stateID;
+    return r;
+  }
+
+}
+
+class Consensus extends Actor {
+  constructor(account,scene) {
+    super();
+    this.account = account;
+    this.id = consensusID
+    this.scene = scene;
+  }
+
+  async transfer(s,fr,to,tok,amt) {
+    const frID = fr.id
+    const toID = to.id
+    const tokID = tok.id
+    return await c.tranfer(s,frID,toID,tokID,amt);
+  }
+
+  async getLedger() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_ledger
+  }
+
+  async getLinearState() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_linstate
+  }
+
+  async getNetworkTime() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_nwtime
+  }
+
+  async getNetworkSeconds() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_nwsecs
+  }
+
+  async getLog() {
+    const g = await c.getStateGlobals(this.scene.stateID)
+    return g.e_messages
+  }
+
+}
+
+class Action {
+  constructor(id,name,owner,scene) {
+    this.id = id;
+    this.name = name;
+    this.owner = owner;
+    this.scene = scene;
+  }
+
+  async resolve(resp,ty="number") {
+    const r = await c.respondWithVal(this.scene.stateID,this.id,resp,this.owner.id,ty)
```

Comment:
You should make it so the values are translated into JS values, perhaps using the CBR representation of the stdlib, so people can see their tuple as a `[]`s and their number as a big number, and so on.